deploy-service/universal/src/main/java/com/pinterest/teletraan/universal/security/AuthZResourceAttributesUtils.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2024 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.teletraan.universal.security;
+
+import com.pinterest.teletraan.universal.security.bean.AuthZResource;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.container.ContainerRequestContext;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = lombok.AccessLevel.PRIVATE)
+public class AuthZResourceAttributesUtils {
+    @SuppressWarnings("unchecked")
+    public static Map<String, String> insertAuthZResourceAttributes(
+            ContainerRequestContext requestContext, AuthZResource.AttributeKeys key, String value) {
+        if (requestContext == null) {
+            return new HashMap<>();
+        }
```

Comment:
Is it valid to call this method with a null `ContainerRequestContext`? Was wondering if it could be considered an error or not

Reply:
Yes, `ContainerRequestContext` is essentially from [Authorizer](https://github.com/dropwizard/dropwizard/blob/8d20ace9b7cc88e2ca5fbc404db4d319c64b71f3/dropwizard-auth/src/main/java/io/dropwizard/auth/Authorizer.java#L33) and it's nullable